flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/common/utils/SchemaTransform.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (C) 2023 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.flink.bigquery.common.utils;
+
+import org.apache.flink.shaded.guava30.com.google.common.collect.ImmutableCollection;
+import org.apache.flink.shaded.guava30.com.google.common.collect.ImmutableMultimap;
+import org.apache.flink.shaded.guava30.com.google.common.collect.Lists;
+
+import com.google.api.services.bigquery.model.TableFieldSchema;
+import com.google.api.services.bigquery.model.TableSchema;
+import com.google.cloud.bigquery.FieldList;
+import org.apache.avro.LogicalType;
+import org.apache.avro.LogicalTypes;
+import org.apache.avro.Schema;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * A utility class that helps on the transformation of BigQuery {@link TableSchema} into Avro {@link
+ * Schema}. Some methods are heavily influenced on the Apache Beam implementation (not externally
+ * accessible methods).
+ */
+public class SchemaTransform {
+
+    static final String NAMESPACE = "org.apache.flink.connector.bigquery";
+    /**
+     * Defines the valid mapping between BigQuery types and native Avro types.
+     *
+     * <p>Some BigQuery types are duplicated here since slightly different Avro records are produced
+     * when exporting data in Avro format and when reading data directly using the read API.
+     */
+    static final ImmutableMultimap<String, Schema.Type> BIG_QUERY_TO_AVRO_TYPES =
+            ImmutableMultimap.<String, Schema.Type>builder()
+                    .put("STRING", Schema.Type.STRING)
+                    .put("GEOGRAPHY", Schema.Type.STRING)
+                    .put("BYTES", Schema.Type.BYTES)
+                    .put("INTEGER", Schema.Type.LONG)
+                    .put("INT64", Schema.Type.LONG)
+                    .put("FLOAT", Schema.Type.DOUBLE)
+                    .put("FLOAT64", Schema.Type.DOUBLE)
+                    .put("NUMERIC", Schema.Type.BYTES)
```

Comment:
Are you referring to Avro logical types here? This class only takes care of translating a BigQuery Table Schema into its Avro representation, for which Avro represent the Decimal type as a BigDecimal encoded data, that's the reason behind of having here a Type.BYTES representation in Avro.
Avro to Flink's Row schemas translation is added later on, and relies entirely on the already existing functionality of the  `org.apache.flink.formats.avro.AvroToRowDataConverters` for data and `org.apache.flink.formats.avro.typeutils.GenericRecordAvroTypeInfo` for schemas.

Reply:
No, I'm referring to the BigQuery type. Also, BigDecimal also has precision and scale and precision

Reply:
I will introduce the change to impose the precision constrain in the AVRO schema transformation and I will test with a live table to ensure it works as expected.

Also, I will explicitly review the Avro schema created by the Storage Read API to see if this constrains are imposed also on the returned data (since the API already generates an Avro schema for the read response).